dotCMS/src/main/java/com/dotcms/filters/interceptor/saml/SamlWebInterceptor.java

```diff
@@ -0,0 +1,557 @@
+                            (String) session.getAttribute(ORIGINAL_REQUEST):
+                            request.getRequestURI();
+
+                    session.removeAttribute(ORIGINAL_REQUEST);
+
+                    Logger.debug(this, ()->  "URI '" + uri + "' belongs to the back-end. Setting the user session data");
+                    session.setAttribute(com.liferay.portal.util.WebKeys.USER_ID, user.getUserId());
+                    session.setAttribute(com.liferay.portal.util.WebKeys.USER,    user);
+                    PrincipalThreadLocal.setName(user.getUserId());
+
+                    renewSession =
+                            this.samlConfigurationService.getConfigAsBoolean(identityProviderConfiguration, SamlName.DOT_RENEW_SESSION)?
+                                this.samlWebUtils.renewSession(request, session): session;
+
+                    this.doAuthenticationLoginSecurityLog(request, identityProviderConfiguration, user);
+                }
+            }
+        }
+
+        return new AutoLoginResult(renewSession, continueFilter);
+    }
+
+    public boolean doLogout(final HttpServletResponse response, final HttpServletRequest request,
+                            final HttpSession session, final IdentityProviderConfiguration identityProviderConfiguration) {
+
+        Logger.debug(this, ()-> "------------------------------ IdP doLogout ---------------------------------");
+
+        final Object nameID           = session.getAttribute(identityProviderConfiguration.getId() + SamlAuthenticationService.SAML_NAME_ID);
+        final String samlSessionIndex = (String) session.getAttribute(identityProviderConfiguration.getId() + SamlAuthenticationService.SAML_SESSION_INDEX);
+        boolean doLogoutDone          = false;
+        Logger.debug(this, ()-> "- idpConfig = " + identityProviderConfiguration);
+        Logger.debug(this, ()-> "- NameID = " + nameID);
+        Logger.debug(this, ()-> "- samlSessionIndex = " + samlSessionIndex);
+
+        try {
+
+            if (null != nameID && null != samlSessionIndex) {
+
+                Logger.debug(this, ()-> "The URI '" + request.getRequestURI() + "' is a logout request. Executing the logout call to SAML");
+                Logger.debug(this, ()-> "Executing dotCMS logout");
+
+                doLogout(response, request);
+
+                Logger.debug(this, ()-> "Executing SAML redirect logout");
+
+                DotSamlProxyFactory.getInstance().samlAuthenticationService().logout(request, response, nameID, samlSessionIndex, identityProviderConfiguration);
+
+                Logger.info(this, ()-> "User '" + nameID + "' has logged out");
+
+                doLogoutDone = true;
+            } else {
+
+                Logger.warn(this, ()->"Couldn't execute the logout request. The SAML NameID or the SAML session index are not in the HTTP session");
+            }
+        } catch (Throwable e) {
```

Comment:
Codacy found an issue: [A catch statement should never catch throwable since it includes errors.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5956529)